sessionctx/variable/varsutil.go

```diff
@@ -43,7 +45,11 @@ func SetDDLReorgWorkerCounter(cnt int32) {
 
 // GetDDLReorgWorkerCounter gets ddlReorgWorkerCounter.
 func GetDDLReorgWorkerCounter() int32 {
-	return atomic.LoadInt32(&ddlReorgWorkerCounter)
+	cnt := atomic.LoadInt32(&ddlReorgWorkerCounter)
+	if cnt > 0 {
+		return cnt
+	}
+	return mathutil.Max(int32(runtime.NumCPU()-2), 1)
```

Comment:
```NumCPU``` will get the wrong result when tidb is in the docker.

Reply:
https://github.com/golang/go/issues/33803

Reply:
@hawkingrei would this be a good solution?

```suggestion
	return mathutil.Max(int32(runtime.GOMAXPROCS(0)-2), 1)
```

```go
package main

import (
	"fmt"
	"runtime"
)

func main() {
	fmt.Printf("NumCPU: %d\n", runtime.NumCPU())
	fmt.Printf("MaxProcs: %d\n", runtime.GOMAXPROCS(0))
}
```

```
$ go run goprocs.go
NumCPU: 8
MaxProcs: 8
```

Reply:
It is a temporary way.

TiDB use [uber-go/automaxprocs](https://github.com/uber-go/automaxprocs) to set the GOMAXPROCS in the startup. it will set it to match the Linux container CPU quota. but it has still weaknesses that it only can set it when to start and the CPU quota can be changed dynamically. so we need to improve it.

Reply:
You can test it by using the ```uber-go/automaxprocs```.